docs/apps/components/sub-form+.md

```diff
@@ -0,0 +1,138 @@
+# Sub-Form+
+
+## Description
+
+Complex layout Sub-Form component for list (array) type form data, supporting nesting of complex layouts.
+
+<img src= "/apps/components/img/subformplus.png">
+
+## Use Case
+
+1. Improved organization
+2. Reduced redundancy
+3. Flexibility and customization
+4. Enhanced data accuracy
+5. Order forms
+6. Employee information forms
+7. Project management forms
+8. Survey forms
+9. Customizable product configuration
+10. Team member contributions
+11. Feedback aggregation
+12. Hierarchical data structures
+
+### Interactivity
+
+1. Show or hide sub-forms based on user choices in the main form.
+2. Dynamically reveal additional sub-forms as users complete previous ones.
+3. Implement branching logic based on sub-form data.
+4. Automatically populate sub-form fields based on data entered in the main form.
+5. Enable data manipulation within sub-forms.
+6. Use sub-forms as dynamic filters or search features.
+7. Integrate drag-and-drop functionality for reordering or prioritizing sub-forms.
+8. Implement collaborative editing for shared sub-forms.
+9. Use progressive disclosure to reveal sub-form content selectively.
+
+## API
+
+### Events
+
+| **Name**| **Description**|
+|---------|----------------|
+|`onChange (element)`| The onChange event occurs when the value of a field is changed|
+|`onRowAdd (element)`|The onRowAdd event occurs when a row is added|
+|`onRowRemove (element)`|The onRowRemove event occurs when a row is removed|
+|`onPageChange (element)`|The onPageChange event occurs when a page is changed|
+
+### Methods
```

Comment:
Add the getValue, getValues and setData methods

Reply:
Added:

|`this.setData`|Set the data in the field. The Value object should be of type { componentId: componentValue }|`(Value: Object)`|
|`this.getValues`|Gets the current values of all fields|`() : Object`|
|`this.getValue`|Get A Value From a component|`(fieldName: String)`|

8. `this.getValues()`
        ```js
        var data = this.getValues();
        console.log(data);
        ```
    
    9. `this.getValue('fieldName')`
        ```js
        var subname = this.getValue('subform_e4cjz3nf');
        console.log('getValue', subname);
        ```
    
    10. `this.setData(Value)`
        ```js
        this.setData({ subform_e4cjz3nf: [ { inputId: '1234', inputId2: 'value2' } ] })